packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -129,12 +129,41 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
     return super.validateCommand();
   }
 
+  void _validateXcodeBuildSettings() {
+    final Map<String, String>? buildSettings = xcodeBuildResult?.xcodeBuildExecution?.buildSettings;
+    if (buildSettings == null) {
+      return;
+    }
+
+    final Map<String, String?> xcodeProjectSettingsMap = <String, String?>{};
+
+    xcodeProjectSettingsMap['iOS App Version'] = buildSettings['MARKETING_VERSION'];
```

Comment:
Are these tittles supposed to be based on something? (XCode setting variable name? Or the name on pubspec? Or the setting name mentioned in https://docs.flutter.dev/deployment/ios#review-xcode-project-settings)
It would be better to have all of them based on the single source of truth if we have one.

But I don't think we have a single place for developers to add/edit all these settings other than XCode?

Reply:
I will use the doc then, since that's what we link to in the error message: 

- version
- build
- display name
- deployment target
- bundle identifier

Reply:
It should be "version number" and "build number" per our docs https://docs.flutter.dev/deployment/ios#update-the-apps-build-and-version-numbers